src/key.rs

```diff
@@ -250,10 +250,8 @@ impl PublicKey {
         }
     }
 
-    #[inline]
```

Comment:
Did you drop this on purpose?

Reply:
I've been dropping these whenever I touch a function, as from previous converstaions that `#[inline]` isn't really needed in many cases. Perhaps it's different here?

Reply:
Yes, I see that point, and I tend to agree but I'm not familiar with the abilities of the rust compiler or  with our best practices here.  @apoelstra
Maybe we should actually document some of the best practices. I'll also help for other rust projects.

Reply:
We have no measured reason to use `#[inline]`, it was just cargo-culted from the rust 0.6-era stdlib. If we later learn there is a benefit we can revisit this but for now it's basically just clutter.